### PR TITLE
Added support for rotation

### PR DESCRIPTION
**Pull request by Arne1303**
Each tile now remembers in which rotation its meant to be laid out, can be used later on to draw the tile in the correct orientation.
I will also issue a pr for Flame which implements this feature in the "tiled_component" class so that every Tile is rotated Correctly.

### CODE REVIEW

File: lib/src/layer.dart [hunk 1]
```diff
@@ -67,9 +70,54 @@ class Layer {
...
+        bool flippedVertically = (globalTileId & FLIPPED_VERTICALLY_FLAG) == FLIPPED_VERTICALLY_FLAG;
+        bool flippedDiagonally = (globalTileId & FLIPPED_DIAGONALLY_FLAG) == FLIPPED_DIAGONALLY_FLAG;
+
+        //Translating the flags into rotation
+        if(
+          !flippedHorizontally && 
+          flippedVertically && 
+          flippedDiagonally){
+            tileRotation[y][x] = 1; //90 °
+        }else if(
```
Comment (luanpotter) [inline lib/src/layer.dart:83]: this is not formatted properly, please do:
```
    if (...) {
```

Comment (luanpotter) [reply lib/src/layer.dart:83]: we can have a method that takes in `globalTileId` and returns the rotation int (1, 2, etc). and here you just do
```
tileRotation[y][x] = foo(globalTileId)
```

File: lib/src/layer.dart [hunk 2]
```diff
@@ -67,9 +70,54 @@ class Layer {
...
+          !flippedHorizontally && 
+          !flippedVertically && 
+          flippedDiagonally){
+            tileRotation[y][x] = 7; //90° + Horizontal flip
+        }else{
+          tileRotation[y][x] = 0; //No rotation
+          /*
+            !flippedHorizontally && 
+            !flippedVertically && 
+            !flippedDiagonally
```
Comment (luanpotter) [inline lib/src/layer.dart:118]: please no commented out code

File: lib/src/layer.dart [hunk 3]
```diff
@@ -97,12 +146,14 @@ class Layer {
...
       
       cols.asMap().forEach((j, Tile t) {
-        tileId = tileMatrix[j][i];
+        tileId = tileMatrix[i][j];
+        rotation = tileRotation[i][j];
         tile = map.getTileByGID(tileId)
-        ..x = i
-        ..y = j
+        ..x = j
+        ..y = i
```
Comment (luanpotter) [inline lib/src/layer.dart:153]: why are you changing i/j x/y on this PR? is this related?

File: lib/src/tile.dart [hunk 4]
```diff
@@ -13,6 +13,19 @@ class Tile {
   int spacing;
   int margin;
 
+  /*
+      Tile Rotation (Clockwise)
```
Comment (luanpotter) [inline lib/src/tile.dart:17]: just curious, did you define this or this a tiled convention?

Comment (Arne1303) [reply lib/src/tile.dart:17]: I did define this, though it was easier to work with a single int value. I could change it, then we would translate it in Tiled Component

File: lib/src/layer.dart (hunk 3 above)
Comment (Arne1303) [reply lib/src/layer.dart:153]: no sry, in my testing the map was rotated by 90° 
I think it just was a flaw in my testing I'll revert that

File: lib/src/tile.dart (hunk 4 above)
Comment (luanpotter) [reply lib/src/tile.dart:17]: Fine by me to keep it, I was just curious on the origin and also if it was based off something else I would ask to add a reference on the comment.

File: lib/src/layer.dart [hunk 5]
```diff
@@ -67,9 +70,54 @@ class Layer {
...
         // Read out the flags
-        var flipped_horizontally = (globalTileId & FLIPPED_HORIZONTALLY_FLAG);
-        var flipped_vertically = (globalTileId & FLIPPED_VERTICALLY_FLAG);
-        var flipped_diagonally = (globalTileId & FLIPPED_DIAGONALLY_FLAG);
+        bool flippedHorizontally = (globalTileId & FLIPPED_HORIZONTALLY_FLAG) == FLIPPED_HORIZONTALLY_FLAG;
+        bool flippedVertically = (globalTileId & FLIPPED_VERTICALLY_FLAG) == FLIPPED_VERTICALLY_FLAG;
+        bool flippedDiagonally = (globalTileId & FLIPPED_DIAGONALLY_FLAG) == FLIPPED_DIAGONALLY_FLAG;
+
+        //Translating the flags into rotation
+        if(
```
Comment (erickzanardo) [inline lib/src/layer.dart:78]: @Arne1303 after a lot of discussion we decided that this code should not be on tiled.dart since this lib is supposed to be kind of a mirror of how the tiled works.

So what tiled.dart must do is to expose these 3 booleans: `flippedHorizontally`, `flippedVertically` and `flippedDiagonally`, instead of the rotation. And the renderer (on our case Flame) should do this logic.

Let us know if you are still interested in continue with this PR, otherwise we can finish it :)

Comment (Arne1303) [reply lib/src/layer.dart:78]: I'm still interested in it :>

File: lib/src/layer.dart [hunk 6]
```diff
@@ -67,12 +70,18 @@ class Layer {
...
+        bool flippedVertically = (globalTileId & FLIPPED_VERTICALLY_FLAG) == FLIPPED_VERTICALLY_FLAG;
+        bool flippedDiagonally = (globalTileId & FLIPPED_DIAGONALLY_FLAG) == FLIPPED_DIAGONALLY_FLAG;
 
-        // Clear the flags
+        // Save rotation flags
+        tileRotation[y][x] = [
+          flippedHorizontally,
+          flippedVertically,
+          flippedDiagonally
+        ];
```
Comment (luanpotter) [inline lib/src/layer.dart:82]: Let's create a class `Flips` that contains three distinct boolean fields? `horizontal`, `vertical` and `diagonal`? It's hard to know what the elements of the array mean when looking at that on other parts of the code.
Then we can have a List<List<Flips>> on the Layer

**Comment by luanpotter** [comment]
Closing in favour of #9 

Don't worry @Arne1303 , your commits are all there
